src/popup.jsx

```diff
@@ -1,16 +1,52 @@
-import React from "react";
+import React, { useEffect,  useState } from "react";
 import { render } from "react-dom";
 import { allWords } from "./helpers.allWords";
-import { getRandomWord } from "./helpers.functions.jsx";
+import { shownWords } from './helpers.shownWords';
+import { getRandomWord } from './helpers.functions.jsx';
 
 const TakeWord = () => {
-  const wordAndTransl = getRandomWord(allWords);
-  return (
-    <div>
-      <h1>{wordAndTransl[0]}</h1>
-      <h2>{wordAndTransl[1]}</h2>
-    </div>
-  );
-};
+    const [wordAndTransl, setWordAndTransl] = useState(getRandomWord(allWords));
+    const [isWordClosed, setIsWordClosed] = useState(false)
+    const [word, translation] = wordAndTransl;
+
+    useEffect(() => {
+        
+    }, [wordAndTransl]);
+
+
+    const takeIt = (shownWords, allWords, word, translation) => {
+        setTimeout(() => {
```

Comment:
This code probably make memory leak, if timeout would be seted and component unmounted. Store timeout in useRef hook 
const timeout = useRef(null);
timeout.current = setTimeout(() => //your timeout ...
 then on useEffect return () => cleatTimeout(timeout.current);